src/uwtools/drivers/forecast.py

```diff
@@ -121,15 +129,92 @@ def requirements(self) -> None:
         ???
         """
 
-    def resources(self) -> None:
+    def resources(self, platform: dict) -> Mapping:
         """
-        ???
+        Set up the formatting for the scheduler batch script.
         """
-
-    def run(self):
+        # Add required fields to platform.
+        # Currently supporting only slurm and none.
+        slurmqueues = ["hera", "jet", "orion", "stampede"]
+        if platform["queue"] is None and platform["machine"] in slurmqueues:
+            platform["queue"] = "slurm"
+        else:
+            platform["queue"] = "none"
+
+        return {
+            "account": platform["account"],
+            "nodes": 1,
+            "queue": platform["queue"],
+            "scheduler": platform["scheduler"],
+            "tasks_per_node": 1,
+            "walltime": "00:01:00",
+        }
+
+    def run(self, log: Optional[Logger] = None) -> None:
         """
         Runs FV3.
         """
+        # Set up logging.
+        if log is None:
+            name = f"{inspect.stack()[0][3]}"
+            log = cli_helpers.setup_logging(
+                log_file="/dev/stdout",
+                log_name=name,
+                quiet=False,
+                verbose=False,
+            )
+        # Read in the config file.
+        forecast_config = config.YAMLConfig(self._config_file)
+
+        # Define forecast model.
+        forecast_model = forecast_config["forecast"]["EXTRN_MDL_NAME"]
+        machine = forecast_config["platform"]["machine"]
+
+        # Prepare directories.
+        run_directory = forecast_config["forecast"]["FCSTDIR"]
+        self.create_directory_structure(run_directory, "delete")
+
+        static_files = forecast_config["forecast"]["STATIC"]
+        self.stage_files(run_directory, static_files, link_files=False)
+        cycledep_files = forecast_config["forecast"]["CYCLEDEP"]
+        self.stage_files(run_directory, cycledep_files, link_files=True)
+
+        # Create the job script.
+        platform = {key: forecast_config["platform"][key] for key in forecast_config["platform"]}
+        platform_resources = self.resources(platform)
+        batch_script = self.batch_script(platform_resources)
+        args = "--export=ALL -n"
```

Comment:
This args statement may not be true on all platforms.

Reply:
Probably a discussion for another place/time, but practice for a long time has been to export NOTHING from the front-end / head nodes to the batch system, but rather to explicitly set any environment variables needed in scripts. The reason is that this avoids the possibility that the same operation mysteriously fails for one user that succeeds for another, or even fails/succeeds at different times, based on whatever might be in the user's shell at the time. Much better IMO is to export nothing, start in a pristine environment on the batch system, and explicitly arrange for precisely the required environment. I've wasted a lot of time helping people troubleshoot nonsense arising from this IMO anti-pattern of exporting shell environments to the batch system. For your consideration, and maybe future discussion.

Reply:
This is pulled from the SRW most common srun arg, so this may be more complicated to change there

Reply:
From today, moving this to "--export=None"